pkg/reconciler/revision/resources/queue.go

```diff
@@ -304,16 +298,56 @@ func makeQueueContainer(rev *v1.Revision, cfg *config.Config) (*corev1.Container
 		}
 	}
 
+	// Sidecar readiness probes
+	multiContainerProbingEnabled := cfg.Features.MultiContainerProbing == apicfg.Enabled
+	readinessProbes := []*corev1.Probe{userContainerReadinessProbe}
+	if multiContainerProbingEnabled {
+		for _, sc := range rev.Spec.GetSidecarContainers() {
+			if sc.ReadinessProbe != nil {
+				var probePort int32
+				switch {
+				case sc.ReadinessProbe.HTTPGet != nil && sc.ReadinessProbe.HTTPGet.Port.IntValue() != 0:
+					probePort = sc.ReadinessProbe.HTTPGet.Port.IntVal
+				case sc.ReadinessProbe.TCPSocket != nil && sc.ReadinessProbe.TCPSocket.Port.IntValue() != 0:
+					probePort = sc.ReadinessProbe.TCPSocket.Port.IntVal
+				case sc.ReadinessProbe.GRPC != nil && sc.ReadinessProbe.GRPC.Port > 0:
+					probePort = sc.ReadinessProbe.GRPC.Port
+				default:
+					return nil, fmt.Errorf("sidecar readiness probe does not define probe port on container: %s", sc.Name)
```

Comment:
sidecars explicitly need to define on what port, we cannot just rely on defaulting to user-port on sidecars.